src/object/obj_class.c

```diff
@@ -94,11 +94,27 @@ daos_oclass_id2name(daos_oclass_id_t oc_id, char *str)
 		strcpy(str, "UNKNOWN");
 		return -1;
 	}
-	if (nr_grps == oc->oc_grp_nr)
+	if (nr_grps == oc->oc_grp_nr) {
 		strcpy(str, oc->oc_name);
-	else
-	/* update oc_name according to nr_grps */
-		strcpy(str, "UNKNOWN");
+	} else {
+		/** update oclass name with group number */
+		char *p = oc->oc_name;
```

Comment:
There are two concerns about this:

1. I think we might need fix caller of daos_oclass_id2name() in some applications, they don't assume str could be MAX_OBJ_CLASS_NAME_LEN.

2. now we support something like this, should we extend daos_oclass_name2id() to support more number of groups as well?

Reply:
About 1, it is a caller-side issue. Normally a 10-byte buffer is currently used. MAX_OBJ_CLASS_NAME_LEN is defined as 24. You can either fix them in this PR or in a separate PR.

Reply:
10-byte might be fine for most of cases.

Reply:
there should be no problem passing a 10 byte buffer here if the oclass fit in those 10 bytes. 24 should fit everything we will have for the next decade i think :-)
ideally the API need to change to pass in the string size, but i don't want to do that here.

Reply:
So the API assumes the client will pass in a sufficiently large buffer?  Seems like it could be improved some.